src/react/2-intro-to-jsx.md

```diff
@@ -0,0 +1,188 @@
+
+### Classes
+
+In JSX, classes are defined using the `className=""` property:
+
+```jsx
+<div className="button primary">So are classes</div>
+```
+
+### Siblings
+
+JSX requires that any sibling elements are wrapped inside of a parent. This is one of the most common errors encountered by beginners.
+
+The following JSX would throw an error:
+
+```jsx
+<div>Sibling 1</div>
+<div>Sibling 2</div>
+```
+
+To fix it, we'd need to wrap them both in a parent:
+
+```jsx
+<div>
+    <div>Sibling 1</div>
+    <div>Sibling 2</div>
+</div>
+```
+
+Now they're both wrapped in a parent `div` and the error will go away!
+
+## JavaScript Interpolation
+
+Because it is integrated into JavaScript, JSX gives us the ability to interpolate JavaScript values into the views we build with it.
+
+```jsx
+  const name = "Bitovi"
+  <div className="button primary">
+    Welcome to {name}!
+  </div>
+```
+
+In the code above, we've used the `{name}` syntax to tell JSX that we want to interpolate the value stored in the `name` variable ("Bitovi") into our view.
+
+You can take this a step furthur by interpolating multiple values, and using JavaScript functions to transform data on the fly. Anything that goes inside `{}` is executed as normal JavaScript.
+
+```jsx
+  const person = {
+    name: "mike",
+    profession: "programmer"
+  }
+  <div className="button primary">
+    <h1>Hi I'm {person.name.toUpperCase()}!</h1>
+    <p>I'm a {person.profession} living in Philadelphia</p>
+  </div>
+```
+
+## Event Handling
+
+One of the most powerful aspects of JavaScript is that it enables developers to respond to events on the browser. With JSX it's really easy to listen for and respond to these events.
+
+
+```jsx 1:3 title="JSX handles user event with ease" subtitle=""
+<div className="button primary">
+  <button onClick={(event) => console.log('Clicked')}>click me</button>
+</div>
+```
+
+In the code above, we've attached an `onClick` listener to the `<button>` element. Whenever this button get's clicked now, the code inside the `onClick={...}` will get executed. This always need to be a function, and that function will get called with an `event` object.
+
+All events supported in vanilla JavaScript are also supported in JSX.
+
+## Components
+
+In React we can store our JSX inside of components. Components are like small containers which can be re-used in your application. For example, you might build a `Button` component which renders our all the JSX required for a button. 
+
+```html
+<div id="root"></div><script crossorigin src="//unpkg.com/react@16/umd/react.development.js"></script><script crossorigin src="//unpkg.com/react-dom@16/umd/react-dom.development.js"></script><script type="jsx">ReactDOM.render(<MyButton />,document.getElementById('root'));
```

Comment:
For whatever reason, this displays weirdly in the online editor. Not sure if it is by design for this initial chunk of code to render all in one line.

Reply:
The long lines of code on these examples are required so that this can run in codepen, I figured reducing it all to one line would be best case, so the user doesn't need to see it